### DISCUSSION TITLE
Black and white theme

### DISCUSSION THREAD

**Discussion by avantgarde0707** [rydmike/flex_color_scheme#248] [Q&A] [answered]
How can I create only black and white theme? Using predefined theme like "High contrast" don't give me expected results. Except error I would like to have everything else in BW colors.

**Comment by rydmike** [comment, marked as answer]
Hi @avantgarde0707,

Thanks for your question. Here are some options.

To make custom color scheme, first go to the last scheme and start adjusting it:

![Screenshot 2024-09-06 at 0 40 16](https://github.com/user-attachments/assets/afca0822-ba49-4cd1-a9f8-5cdce2e06a9d)

For a black and white theme, make primary black in light mode:

![Screenshot 2024-09-06 at 0 44 23](https://github.com/user-attachments/assets/239d64e0-6050-450f-833c-a9265dd4baa0)

Then if we are lazy, choose to use only primary color, this will compute the other colors as grey scale versions (when not using a seeded scheme):


![Screenshot 2024-09-06 at 0 44 49](https://github.com/user-attachments/assets/7de0749c-5f1b-458c-8b11-d25b1999b47d)

That pretty much gives you a black and white grey scale light theme already with just primary defined as black.

![Screenshot 2024-09-06 at 0 45 55](https://github.com/user-attachments/assets/63a42562-0e55-4c21-b270-16b6dae92eb2)

Sure it has grey scale for other colors, you can of course opt to keep input colors at 6 colors and defined all the other main colors as pure black too, or other darker greys. 

You will need some monochrome greys in there for some widgets to look nice, typically on container colors, just black and white will not be so brilliant for some widgets, like eg Chips that gets faulty contrast color then, but sure doable, here just to show it:

![Screenshot 2024-09-06 at 0 50 26](https://github.com/user-attachments/assets/debdb1ee-2d29-4dc4-b619-4eaa0b35eb74)

You can also use these colors as input color to make a "Seed Generated" balanced greyscale Colorscheme that gets more usable greys without any chroma. Let's try that too.

We need to use all key colors, to seed, our blacks for primary, secondary and tertiary, then lock them to their black inputs and we should also use the **chroma** seed generation, which since inputs have no chroma will then become greyscale.

We also need to set surface tint to white, and we can lock a bunch of other tones black/white too, or keep them to greyscale tones if you want. Here I changed them black/whites for more contrasty look with surface and on colors:

![Screenshot 2024-09-06 at 0 54 40](https://github.com/user-attachments/assets/1aa798c2-c13f-4bf6-9898-5ada5178f04b)

This gives us a "black and white" theme that has good greys that work well with Material-3 component's default color mappings from the Colorscheme:

![Screenshot 2024-09-06 at 0 58 52](https://github.com/user-attachments/assets/39e18126-95f8-44a8-8be9-862a735c71e2).


Obviously for the dark theme we will need to do the same, but using **white** as our main color:

Same idea:

![Screenshot 2024-09-06 at 1 04 19](https://github.com/user-attachments/assets/4e09a7d0-ae1f-43b7-b65e-ca6b87845acc)

This gives us pretty decent "black and white" M3 widgets:

![Screenshot 2024-09-06 at 1 04 52](https://github.com/user-attachments/assets/9887fa17-2a81-4e8d-88a4-0b3a87bb9991)

---

For some components that by default map to container colors that are now greys, you can change them to use eg Primary, for example the FAB:

![Screenshot 2024-09-06 at 1 20 52](https://github.com/user-attachments/assets/d0747b2a-aa5a-4851-a54a-f1f35a7c0c98)
![Screenshot 2024-09-06 at 1 21 38](https://github.com/user-attachments/assets/0b3b2011-bd34-425b-9056-548c94966840)
![Screenshot 2024-09-06 at 1 21 52](https://github.com/user-attachments/assets/6a26c254-bf03-4309-bf7c-27140927ba9d)


---

In coming version 8, there will be some built in quick choice to produce a nice monochrome theme like above, but it is already very doable, just a bit tricky.

A monochrome theme can be very nice for protyping without any colors, so e.g. test users focus on features, layout, and interactions first and not the bloody colors 😄

**Reply by avantgarde0707** [reply]
Wow, thank you for this update. Looks like exactly what I'm looking for. Will try. Amazing

**Reply by avantgarde0707** [reply]
Looks very good and exactly what I'm looking for. Thank you for this explanation and for your great work here!

**Reply by rydmike** [reply]
By the way and for info FCS 8.0.0 is out as early access as 8.0.0-dev.1

For convenience it has a black and white theme, one with more grey tones, and one with "sepia" tones a brown/yellow warm "monochrome" theme. 

**FlexColorScheme** package dev release 8.0.0-dev.1 has been published. This is a BIG update with breaking changes and many new features. It has its own dev pre-release of the [**Themes Playground**](https://rydmike.com/flexcolorscheme/themesplayground-dev) as well, feel free to take a look. No new API changes are planned for the stable 8.0.0 release from this dev.1 release.

We would appreciate any feedback about found issues from early adopters. 

* [Doc site has some pre-info](https://docs.flexcolorscheme.com/)
* [Dev release of PKG here](https://pub.dev/packages/flex_color_scheme/versions/8.0.0-dev.1)
* [The PKG changelog has the most detailed info](https://pub.dev/packages/flex_color_scheme/versions/8.0.0-dev.1/changelog)

**FOR INFO** there will be no new features or changes in it compared to this one. Only any issues found from the quality assurance will be addressed. The stable release will also include many doc updates, but docs site will continue to be edited and revised AFTER the 8.0.0 stable release.

**Reply by avantgarde0707** [reply]
That looks nice! Will try it. Thanks for your update!!!

**Reply by rydmike** [reply]
And now it is released as 8.0.0 stable

- https://github.com/rydmike/flex_color_scheme
- https://rydmike.com/flexcolorscheme/themesplayground-latest/
- https://docs.flexcolorscheme.com/whats_new
